crd/apis/gcpfirewall/v1alpha1/types.go

```diff
@@ -82,20 +82,46 @@ type GCPFirewallSpec struct {
 	// +optional
 	Ports []ProtocolPort `json:"ports,omitempty"`
 
-	// A collection of sources from which ingress traffic is allowed. If this
-	// field is missing, no ingress rules are programmed. if this field is empty,
+	// A collection of sources and destinations to determine which ingress traffic is allowed.
+	// If this field is missing, no ingress rules are programmed.
+	// +optional
+	Ingress *GCPFirewallIngress `json:"ingress,omitempty"`
+}
+
+// GCPFirewallIngressPeer describes a peer to allow traffic from. Items in each of the
```

Comment:
This needs to be updated, right?.

Maybe something like:
```
// GCPFirewallIngress describes a source and a destination for the ingress firewall rule. 
// If source is nil or empty, the traffic is allowed from all sources (0.0.0.0/0).
// If destination is nil or empty, the traffic is allowed to all kubernetes cluster entities
// (nodes, pods and services) from the specified sources.
// If both are nil, the traffic is allowed from all sources (0.0.00/0) to the cluster entities.
```

Reply:
Added to line 86 as it will show up in the yaml.